contracts/access/manager/AccessManager.sol

```diff
@@ -406,7 +443,10 @@ contract AccessManager is Context, Multicall, IAccessManager {
     /**
      * @dev Internal version of {setRoleAdmin} without access control.
      *
-     * Emits a {RoleAdminChanged} event
+     * Emits a {RoleAdminChanged} event.
+     *
+     * NOTE: Setting the admin role as the `PUBLIC_ROLE` is allowed, but it will effectively allow
```

Comment:
Do we want to prevent that ? (same question for `_setRoleGuardian`)

Reply:
I'd personally prevent both because I don't see why anyone would do that but we need to analyze a bit better.
We can do it in the tests PR #4613